### PR TITLE
Add Exclude Parameter to Nearest

### PR DESCRIPTION
**Pull request by UrbanGISer**

### CODE REVIEW

File: knime_extension/src/nodes/spatialtool.py [hunk 1]
```diff
@@ -730,7 +738,8 @@ def get_default(cls):
         label="Return only geometries of the same geometry type",
         description="""If selected, the node returns only geometries of the same geometry type as the geometry column 
         from the left (top) input table, otherwise it returns all resulting geometries.""",
-        default_value=lambda v: True if v < knext.Version(1, 1, 0) else False,
+        # default_value=lambda v: True if v < knext.Version(1, 1, 0) else False,
```
Comment (koettert) [inline knime_extension/src/nodes/spatialtool.py:741]: Why did you change this? This should stay the same to be backward compatible otherwise older workflows will behave differently after the update.

File: knime_extension/src/nodes/spatialtool.py [hunk 2]
```diff
@@ -461,6 +461,13 @@ class NearestJoinNode2:
 
     column_selection = ColumnSelection()
 
+    exclusive_equal = knext.BoolParameter(
+        label="Exclude the nearest geometries that are equal to the input geometry ",
+        description="""If checked, the nearest geometries that are equal to the input geometry will not be returned.""",
+        default_value=lambda v: True if v < knext.Version(1, 1, 0) else False,
+        since_version="1.1.0",
```
Comment (koettert) [inline knime_extension/src/nodes/spatialtool.py:468]: Shouldn't the version be 1.3.0 in this line and the above one?

**Review by koettert — CHANGES_REQUESTED** [review]

File: knime_extension/src/nodes/spatialtool.py (hunk 1 above)
Comment (UrbanGISer) [reply knime_extension/src/nodes/spatialtool.py:741]: I would like to set False  always as the default, which is compatible with the older workflow

Comment (koettert) [reply knime_extension/src/nodes/spatialtool.py:741]: True the parameter was introduced with 1.1.0 with false as default

File: knime_extension/src/nodes/spatialtool.py (hunk 2 above)
Comment (koettert) [reply knime_extension/src/nodes/spatialtool.py:468]: Set since version to 1.3.0 in order to tell KNIME that this is a new parameter. This way existing workflows with use the default value.
Set the _default_value_ to false since it was always false and should be false by default.

Comment (UrbanGISer) [reply knime_extension/src/nodes/spatialtool.py]: Revised. Be aware of the 1.3 version of 1.3 while testing
